.konflux/overlay/overlay.bash

```diff
@@ -0,0 +1,356 @@
+
+    local parsed=$(getopt --options="$options" --longoptions="$long_options" --name "$SCRIPT_NAME" -- "$@")
+    eval set -- "$parsed"
+
+    local map_staging=0
+    local map_production=0
+    declare -g ARG_MAPPING_FILE=""
+    declare -g ARG_PINNING_FILE=""
+    declare -g ARG_CSV_FILE=""
+    declare -g ARG_MAP=""
+    while true; do
+        case $1 in
+            --help)
+                usage
+                exit
+                ;;
+            --set-csv-file)
+                ARG_CSV_FILE=$2
+                shift 2
+                ;;
+            --set-pinning-file)
+                ARG_PINNING_FILE=$2
+                shift 2
+                ;;
+            --set-mapping-file)
+                ARG_MAPPING_FILE=$2
+                shift 2
+                ;;
+            --set-mapping-staging)
+                map_staging=1
+                ARG_MAP=$MAP_STAGING
+                shift 1
+                ;;
+            --set-mapping-production)
+                map_production=1
+                ARG_MAP=$MAP_PRODUCTION
+                shift 1
+                ;;
+            --)
+                shift
+                break
+                ;;
+            *)
+                echo "Error: unexpected option: $1" >&2
+                usage
+                exit 1
+                ;;
+        esac
+    done
+
+    # validate images file
+    if [[ -n $ARG_PINNING_FILE && ! -f "$ARG_PINNING_FILE" ]]; then
+        echo "Error: file '$ARG_PINNING_FILE' does not exist." >&2
+        exit 1
+    fi
+
+    # validate csv file
+    if [[ -n $ARG_CSV_FILE && ! -f "$ARG_CSV_FILE" ]]; then
+        echo "Error: file '$ARG_CSV_FILE' does not exist." >&2
+        exit 1
+    fi
+
+    # validate map options
+    if [[ $map_staging -eq 1 && $map_production -eq 1 ]]; then
+        echo "Error: cannot specify both '--set-mapping-staging' and '--set-mapping-production'." >&2
+        exit 1
+    fi
+
+    if [[ $map_staging -eq 1 || $map_production -eq 1 ]]; then
+        if [[ ! -n $ARG_MAPPING_FILE ]]; then
+            echo "Error: specify '--set-mapping-file' to use a container registry map file." >&2
+            exit 1
+        fi
+
+        if [[ ! -f "$ARG_MAPPING_FILE" ]]; then
+            echo "Error: file '$ARG_MAPPING_FILE' does not exist." >&2
+            exit 1
+        fi
+    fi
+
+    if [[ -n $ARG_MAPPING_FILE ]]; then
+        if [[ $map_staging -eq 0 && $map_production -eq 0 ]]; then
+            echo "Error: specify '--set-mapping-staging' or '--set-mapping-production'." >&2
+            exit 1
+        fi
+    fi
+
+    echo "Parsing args completed!"
+}
+
+overlay_release()
+{
+    echo "Overlaying relase..."
+
+    local display_name="lifecycle-agent-operator"
+    local description="lifecycle-agent-operator"
+    local version="4.20.0"
+    local name="lifecycle-agent-operator"
+    local name_version="$name.v$version"
+    local manager="lifecycle-agent-operator"
```

Comment:
were is `manager` used here?

Reply:
I could not identify it being used anywhere, but I wanted to keep the script as in-tact as possible versus the oran implementation from @rauhersu